tests/unit/python/foglamp/services/core/api/test_common_ping.py

```diff
@@ -17,46 +19,90 @@
+                assert 1 == content_dict["dataPurged"]
+                assert content_dict["authenticationOptional"] is True
+                log_params = 'Received %s request for %s', 'GET', '/foglamp/ping'
+                logger_info.assert_called_once_with(*log_params)
 
 @pytest.allure.feature("unit")
 @pytest.allure.story("api", "common")
-async def test_shutdown_http(test_server, test_client, loop):
-    app = web.Application()
-    # fill route table
-    routes.setup(app)
+async def test_ping_http_auth_required(test_server, test_client, loop):
+    payload = {"return": ["key", "description", "value"], "sort": {"column": "key", "direction": "asc"}}
+    result = {"rows": [
+                {"value": 1, "key": "PURGED", "description": "blah6"},
+                {"value": 2, "key": "READINGS", "description": "blah1"},
+                {"value": 3, "key": "SENT_1", "description": "blah2"},
+                {"value": 4, "key": "SENT_2", "description": "blah3"},
+                {"value": 5, "key": "SENT_3", "description": "blah4"},
+                {"value": 6, "key": "SENT_4", "description": "blah5"},
+               ]}
 
-    server = await test_server(app)
-    server.start_server(loop=loop)
+    mockedStorageClient = MagicMock(StorageClient)
+    with patch.object(middleware._logger, 'info') as logger_info:
+        with patch.object(connect, 'get_storage', return_value=mockedStorageClient):
+            with patch.object(mockedStorageClient, 'query_tbl_with_payload', return_value=result) as query_patch:
+                app = web.Application(loop=loop, middlewares=[middleware.auth_middleware])
+                # fill route table
+                routes.setup(app)
 
-    client = await test_client(server)
-    resp = await client.put('/foglamp/shutdown', data=None)
-    assert 200 == resp.status
-    content = await resp.text()
-    content_dict = json.loads(content)
-    assert "FogLAMP shutdown has been scheduled. Wait for few seconds for process cleanup." == content_dict["message"]
+                server = await test_server(app)
+                server.start_server(loop=loop)
+
+                client = await test_client(server)
+                # note: If the parameter is app aiohttp.web.Application
+                # the tool creates TestServer implicitly for serving the application.
+                resp = await client.get('/foglamp/ping')
+                assert 200 == resp.status
+                content = await resp.text()
+                content_dict = json.loads(content)
+                assert 0.0 < content_dict["uptime"]
+                assert 2 == content_dict["dataRead"]
+                assert 18 == content_dict["dataSent"]
+                assert 1 == content_dict["dataPurged"]
+                assert content_dict["authenticationOptional"] is False
+                log_params = 'Received %s request for %s', 'GET', '/foglamp/ping'
+                logger_info.assert_called_once_with(*log_params)
```

Comment:
Please add query_patch.assert_called_once_with(payload) and fix indentation for logger_ info